### ISSUE TITLE
Bad state:Can't finalize a finalized MultipartFile

### ISSUE THREAD

**Issue by Shkryabminer** [cfug/dio#1606]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

I try to login with email/password in my App. If  the internet connection is lost, I try to Switch On Wifi Connection and resend request in Dio Error Interceptor:

return handler
              .resolve(await _dio.request(options.path,
              queryParameters: options.queryParameters,
              data: options.data,
              options: dio.Options(
                method: options.method,
                headers: options.headers,
                extra: options.extra,
                contentType: options.contentType,
              )));
But I get : Bad state:Can't finalize a finalized MultipartFile error

this is data that I post to server
FormData formData =
        new FormData.fromMap({"email": email, "password": pass});
    final response = await _dio.post(
      "/Api/Login/checklogin", data: formData);
